rosbag2_transport/test/rosbag2_transport/test_qos.cpp

```diff
@@ -58,13 +59,14 @@ TEST(TestQoS, supports_version_4)
     "  avoid_ros_namespace_conventions: false\n";
 
   YAML::Node loaded_node = YAML::Load(serialized_profiles);
-  auto deserialized_profiles = loaded_node.as<std::vector<rosbag2_transport::Rosbag2QoS>>();
+  auto deserialized_profiles = YAML::decode_for_version<std::vector<rosbag2_storage::Rosbag2QoS>>(
+    loaded_node, 8);
```

Comment:
It would be nice to make it work and check when using the default `loaded_node.as<std::vector<rosbag2_transport::Rosbag2QoS>>();` for version 4